src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileCreator.java

```diff
@@ -228,6 +230,35 @@ public void updateInputRequestWithLatestSidamUserInfo(UserProfileCreationData pr
         }
     }
 
+    private void updateNamesInSidam(UserProfileCreationData profileData, IdamRolesInfo idamRolesInfo, String origin) {
+        if (ORIGIN_SRD.equals(origin)) {
+            updateSidamUserInfoWithUserProfileDetails(profileData, idamRolesInfo);
+
+        } else {
+            // set updated user info
+            updateInputRequestWithLatestSidamUserInfo(profileData, idamRolesInfo);
+        }
+    }
+
+    private void updateSidamUserInfoWithUserProfileDetails(UserProfileCreationData profileData,
+                                                          IdamRolesInfo idamRolesInfo) {
+        profileData.setStatus(IdamStatusResolver.resolveIdamStatus(idamRolesInfo));
+        if (!(profileData.getFirstName().equals(idamRolesInfo.getForename()) && profileData.getLastName()
+                .equals(idamRolesInfo.getSurname()))) {
+            UpdateUserDetails updateUserDetails = new UpdateUserDetails(profileData.getFirstName(),
+                    profileData.getLastName(), "ACTIVE".equals(profileData.getStatus().toString()));
+            AttributeResponse attributeResponse = idamService
+                    .updateUserDetails(updateUserDetails, idamRolesInfo.getId());
+            HttpStatus status = HttpStatus.valueOf(attributeResponse.getIdamStatusCode());
+            if (!status.is2xxSuccessful()) {
+                log.error("{}:: failed sidam update names as per userprofile POST call for the given origin",
```

Comment:
we can say
` log.error("{}:: failed to update attributes in sidam. Reason {}",
                        loggingComponentName, attributeResponse.getIdamMessage());`